plugins/inputs/t128_lte/t128_lte.go

```diff
@@ -0,0 +1,384 @@
+					Name string `json:"name"`
+				} `json:"node"`
+			} `json:"router"`
+		} `json:"authority"`
+	} `json:"config"`
+}
+
+type t128Config struct {
+	Config struct {
+		Authority struct {
+			Router []struct {
+				Node []struct {
+					DeviceInterface []lteDeviceInterface
+				} `json:"node"`
+			} `json:"router"`
+		} `json:"authority"`
+	} `json:"config"`
+}
+
+type lteDeviceInterface struct {
+	Name            string `json:"name"`
+	Type            string `json:"type"`
+	TargetInterface string `json:"targetInterface"`
+}
+
+type lteDeviceState struct {
+	LTE struct {
+		Active3GPPProfile struct {
+			APN string `json:"APN"`
+		} `json:"Active 3GPP Profile"`
+		SignalStrength     string `json:"Signal Strength"`
+		RSRQSignalStrength string `json:"RSRQ Signal Strength"`
+		RSSISignalStrength string `json:"RSSI Signal Strength"`
+		RadioInterface     string `json:"Radio Interface"`
+		SNRSignalStrength  string `json:"SNR Signal Strength"`
+		RSRPSignalStrength string `json:"RSRP Signal Strength"`
+		RegistrationStatus string `json:"Registration Status"`
+		ActiveBandClass    string `json:"Active Band Class"`
+		Carrier            string `json:"Carrier"`
+		ConnectionStatus   string `json:"Connection Status"`
+		Failure            struct {
+			Reason string `json:"reason"`
+		} `json:"failure"`
+	} `json:"LTE"`
+}
+
+type lteDeviceInfo struct {
+	DeviceInfo struct {
+		IMEI  string `json:"imei"`
+		ICCID string `json:"iccid"`
+		IMSI  string `json:"imsi"`
+	} `json:"Device Info"`
+}
+
+// T128Lte is an input for metrics of a 128T router instance
+type T128Lte struct {
+	CollectorName  string `toml:"collector_name"`
+	RouterName     string `toml:"router_name"`
+	NodeName       string `toml:"node_name"`
+	T128ConfigFile string `toml:"t128_config_file"`
+	LTEPathDir     string `toml:"lte_path_dir"`
+	GlobalInitPath string `toml:"global_init_path"`
+	LocalInitPath  string `toml:"local_init_path"`
+}
+
+// SampleConfig returns the default configuration of the Input
+func (*T128Lte) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns a one-sentence description on the Input
+func (*T128Lte) Description() string {
+	return "Make a 128T LTE query and return the data"
+}
+
+func (plugin *T128Lte) Init() error {
+	var err error
+	plugin.RouterName, err = plugin.loadRouterName()
+	if err != nil {
+		return err
+	}
+	plugin.NodeName, err = plugin.loadNodeName()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (plugin *T128Lte) Gather(acc telegraf.Accumulator) error {
+	var found bool
+	found = false
+	for lteInterface := range plugin.getLteInterfaces() {
+		found = true
+		lteMetric, err := plugin.getLteStateInfo(lteInterface, plugin.LTEPathDir)
+		lteInfo, err := plugin.getLteInfo(lteInterface, plugin.LTEPathDir)
+		if err != nil {
+			acc.AddError(fmt.Errorf("Unable to find LTE info %w", err))
+		} else {
+			name := lteInterface.Name
+			collectorName := plugin.CollectorName
+			if !hasRequiredData(lteMetric) {
```

Comment:
Do this immediately after `getLteStateInfo`. If the data will be dropped, don't bother executing other code before the check.